internal/catalog/internal/controllers/endpoints/controller_test.go

```diff
@@ -680,6 +685,37 @@ func (suite *controllerSuite) TestController() {
 	// Verify that the endpoints were not regenerated
 	suite.client.RequireVersionUnchanged(suite.T(), endpointsID, endpoints.Version)
 
+	// Update the service.
+	updatedService := rtest.Resource(types.ServiceType, "api").
```

Comment:
Adding one more case for when service is updated. This wasn't catching any bugs, but I added it to test some behavior and decided to keep for extra coverage.